src/Interpreters/Access/InterpreterGrantQuery.cpp

```diff
@@ -178,6 +178,21 @@ namespace
                 elements_to_revoke.emplace_back(std::move(element_to_revoke));
         }
 
+        /// Additional check for REVOKE
+        ///
+        /// If user1 has the rights
+        /// GRANT SELECT ON *.* TO user1;
+        /// REVOKE SELECT ON system.* FROM user1;
+        /// REVOKE SELECT ON mydb.* FROM user1;
+        ///
+        /// And user2 has the rights
+        /// GRANT SELECT ON *.* TO user2;
+        /// REVOKE SELECT ON system.* FROM user2;
+        ///
+        /// the query `REVOKE SELECT ON *.* FROM user1` executed by user2 should succeed.
+        if (current_user_access.getAccessRights()->containsWithGrantOption(access_to_revoke))
+            return;
```

Comment:
What is the point of calling `current_user_access.checkGrantOption(elements_to_revoke)` at the end of this function now? I mean it seems you've already checked everything in `containsWithGrantOption()`.

Reply:
Actually, the only reason is that I want to reuse the nice exception message from the `checkGrantOption` function.

Reply:
Let's move throwing of this nice exception message to a separate function then maybe?